bionic/cache.py

```diff
@@ -494,25 +494,47 @@ def _exact_metadata_url_for_query(self, query):
     def _load_metadata_from_url(self, url):
         try:
             metadata_yaml = self._fs.read_bytes(url).decode('utf8')
-            return ArtifactMetadataRecord.from_yaml(metadata_yaml)
+            record = ArtifactMetadataRecord.from_yaml(metadata_yaml)
```

Comment:
This is pretty reasonable. I wonder if it would be cleaner for this logic to happen inside the `ArtifactMetadataRecord` code? (Like within `from_yaml` and `from_content`?) It's a little annoying to have to pass in the metadata URL but it also keeps the record object immutable and saves us from having to remember to do this if we load a record anywhere else.

Reply:
Good point about immutability. I missed that since immutability isn't enforced here.